src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -527,18 +529,65 @@ public String getLaunchURLForCaseId(final UUID caseId, LaunchRequestDTO requestP
 
     CaseContainerDTO caseDetails = getLaunchCase(caseId);
 
+    SingleUseQuestionnaireIdDTO newQuestionnaireIdDto =
+        getNewQidForCase(caseDetails, requestParamsDTO.getIndividual());
+
+    String questionnaireId = newQuestionnaireIdDto.getQuestionnaireId();
+    String formType = newQuestionnaireIdDto.getFormType();
+
+    String eqUrl = createLaunchUrl(formType, caseDetails, requestParamsDTO, questionnaireId);
+    publishSurveyLaunchedEvent(caseDetails.getId(), questionnaireId, requestParamsDTO.getAgentId());
+    return eqUrl;
+  }
+
+  @Override
+  public UACResponseDTO getUACForCaseId(UUID caseId, UACRequestDTO requestParamsDTO)
+      throws CTPException {
+    log.with("caseId", caseId)
+        .with("request", requestParamsDTO)
+        .debug("Processing request to get UAC for Case");
+
+    CaseContainerDTO caseDetails = getLaunchCase(caseId);
+
+    SingleUseQuestionnaireIdDTO newQuestionnaireIdDto =
+        getNewQidForCase(caseDetails, requestParamsDTO.getIndividual());
+
+    UACResponseDTO response =
+        UACResponseDTO.builder()
+            .id(newQuestionnaireIdDto.getQuestionnaireId())
+            .uac(newQuestionnaireIdDto.getUac())
+            .dateTime(DateTimeUtil.nowUTC())
+            .build();
+
+    return response;
+  }
+
+  /**
+   * Request a new questionnaire Id for a Case
+   *
+   * @param caseDetails of case for which to get questionnaire Id
+   * @param individual whether request for individual questionnaire
+   * @return
+   */
+  private SingleUseQuestionnaireIdDTO getNewQidForCase(
+      CaseContainerDTO caseDetails, boolean individual) throws CTPException {
+
     CaseType caseType = CaseType.valueOf(caseDetails.getCaseType());
     if (!(caseType == CaseType.CE || caseType == CaseType.HH || caseType == CaseType.SPG)) {
       throw new CTPException(Fault.BAD_REQUEST, "Case type must be SPG, CE or HH");
     }
 
-    UUID individualCaseId = createIndividualCaseId(caseType, caseDetails, requestParamsDTO);
+    UUID parentCaseId = caseDetails.getId();
+    UUID individualCaseId = null;
+    if (caseType == CaseType.HH && individual) {
+      caseDetails = createIndividualCase(caseDetails);
```

Comment:
I would probably use a different variable here, as the reponsibility of caseDetails is to carry a value into the method and then we are reassigning it.....

Reply:
Actually thought quite a bit about that very point. Originally there was a method:
` private UUID createIndividualCaseId(CaseType caseType, CaseContainerDTO caseDetails, LaunchRequestDTO requestParamsDTO)`
The changing of the caseDetails instance to an individual case was completely hidden in this method, not referred to in the method name or returned as the 'new' individual case. So I renamed it createIndividualCase and made the logic explicit in the method that calls it but, it is not a deep copy of the original case returning a different instantiation in the heap, that seems wasteful so is the same parent HH, CE, or SPG case amended now to be an individual case. I wanted to show the variable caseDetails now is not the original case any longer to be sure no one used  it as such.
Changing the variable name would also add an unnecessary assignment to null to make it available in the outer code block, really createIndividualCase does not have to return it at all as before for it still to be available in the calling method, it was just to make explicit what has happened.

Reply:
If you are happy with that then that's fine by me, and I can see both viewpoints.